### PR TITLE
DOC: Clear warnings when building docs

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

This doc clears some warnings thrown by sphinx when building the documentation for sklearnex.

The  following warnings should now disappear:
```
<path>/scikit-learn-intelex/doc/sources/kaggle/classification.rst:18: WARNING: Duplicate explicit target name: "support vector classification (svc) for multiclass classification".
looking for now-outdated files... none found
pickling environment... done
checking consistency... <path>/scikit-learn-intelex/doc/sources/samples/daal4py_data_science.ipynb: WARNING: document isn't included in any toctree
<path>/scikit-learn-intelex/doc/sources/samples/tsne.ipynb: WARNING: document isn't included in any toctree
done
preparing documents... WARNING: unsupported theme option 'display_version' given
```

Note: here I'm switching Sphinx option `display_version` to `version_selector`, as suggested in their docs:
https://sphinx-rtd-theme.readthedocs.io/en/stable/configuring.html#confval-display_version
image

If I understand the docs correctly, this `display_version` option shouldn't actually be doing anything when building with the version of Sphinx in the locked requirements, but they mention that this option requires the ReadTheDocs add-on, which I don't know if is used by the CI job that builds the docs:
https://sphinx-rtd-theme.readthedocs.io/en/stable/configuring.html#confval-version_selector
image

---

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.

**Performance**

Not applicable

### CODE REVIEW

**Comment by icfaust** [comment]
I'm going to step back from reviewing this one because I don't have a good handle on the docs. Sorry.

File: doc/sources/conf.py [hunk 1]
```diff
@@ -128,7 +128,7 @@
 
 html_theme_options = {
     "logo_only": False,
-    "display_version": True,
+    "version_selector": True,
```
Comment (Vika-F) [inline doc/sources/conf.py:131]: Maybe it would be better to remove it at all? Because:

1.  Looks like `version_selector` won't work correctly anyway as our docs are not hosted at Read the Docs,
2. Our docs are not organized in the way to provide information for different versions. The information is only provided for the latest version anyway.

File: doc/sources/samples.rst [hunk 2]
```diff
@@ -53,13 +53,21 @@ If you want to run them locally, refer to `these instructions
...
+   Intel® Extension for Scikit-learn TSNE example <samples/tsne.ipynb>
+
 .. rubric:: Jupyter Notebook Samples
 
 .. toctree::
    :maxdepth: 1
 
    Intel® Extension for Scikit-Learn* Getting Started Sample <https://github.com/oneapi-src/oneAPI-samples/tree/master/AI-and-Analytics/Getting-Started-Samples/Intel_Extension_For_SKLearn_GettingStarted>
    Intel® Extension for Scikit-Learn*: SVC for Adult dataset Performance Sample <https://github.com/oneapi-src/oneAPI-samples/tree/master/AI-and-Analytics/Features-and-Functionality/Intel_Extension_For_SKLearn_Performance_SVC_Adult>
+   Utilizing daal4py in Data Science Workflows <samples/daal4py_data_science.ipynb>
```
Comment (Vika-F) [inline doc/sources/samples.rst:70]: We are moving towards sklearnex estimators interface. We are not going to extend the legacy daal4py API.
I think it's better not to extend the docs with the additional daal4py samples.

It would be better to update the notebook using sklearnex API and then add the link to it.

**Review by Vika-F — COMMENTED** [review]
Thanks for fixing the warnings!
Please take a look at the comments below.

File: doc/sources/conf.py (hunk 1 above)
Comment (david-cortes-intel) [reply doc/sources/conf.py:131]: I think it refers to this feature, which is working correctly in _some_ versions of the docs:
2024-10-28 13 03 16 intel github io b05110fb1274

File: doc/sources/samples.rst (hunk 2 above)
Comment (david-cortes-intel) [reply doc/sources/samples.rst:70]: Then I guess removing those .ipynb files might be a better option.

Comment (david-cortes-intel) [reply doc/sources/samples.rst]: Left that `daal4py_data_science.ipynb` out from the docs and cleared the warning instead by not copying to the samples folder.

File: doc/build-doc.sh [hunk 3]
```diff
@@ -26,6 +26,9 @@ mkdir $SAMPLES_DIR
 # copy jupyter notebooks
 cd ..
 cp examples/notebooks/*.ipynb doc/$SAMPLES_DIR
+if [[ -f "doc/$SAMPLES_DIR/daal4py_data_science.ipynb" ]]; then
+    rm "doc/$SAMPLES_DIR/daal4py_data_science.ipynb"
+fi
```
Comment (Vika-F) [inline doc/build-doc.sh:31]: It would be better not to copy the file at all.
Something like that should work:
```suggestion
rsync -a --exclude='doc/$SAMPLES_DIR/daal4py_data_science.ipynb' examples/notebooks/*.ipynb doc/$SAMPLES_DIR
```

File: doc/sources/conf.py (hunk 1 above)
Comment (david-cortes-intel) [reply doc/sources/conf.py:131]: It actually shows more entries if you click it:
image

**Review by Vika-F — APPROVED** [review]
Let's get green CI and merge.
